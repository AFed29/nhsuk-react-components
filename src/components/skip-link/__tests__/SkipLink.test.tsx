import React, { createRef } from 'react';
import { shallow } from 'enzyme';
import SkipLink from '..';

describe('SkipLink', () => {
  it('matches snapshot', () => {
    const component = shallow(<SkipLink />);
    expect(component).toMatchSnapshot('SkipLink');
    component.unmount();
  });

  it('does not add focusTargetRef or disableDefaultBehaviour as attributes when rendered', () => {
    const ref = createRef<HTMLElement>();
    const component = shallow(<SkipLink focusTargetRef={ref} disableDefaultBehaviour aria-label="test" />);
    expect(component.prop('focusTargetRef')).toBeUndefined();
    expect(component.prop('disableDefaultBehaviour')).toBeUndefined();
    expect(component.prop('aria-label')).toEqual('test');
    component.unmount();
  });
});
