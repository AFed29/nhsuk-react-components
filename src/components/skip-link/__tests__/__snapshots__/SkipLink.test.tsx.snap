// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink matches snapshot: SkipLink 1`] = `
<a
  className="nhsuk-skip-link"
  href="#maincontent"
  onClick={[Function]}
>
  Skip to main content
</a>
`;
